core/module.cc

```diff
@@ -16,13 +16,80 @@
 const Commands Module::cmds;
 
 std::map<std::string, Module *> ModuleBuilder::all_modules_;
+std::map<std::string, Node> ModuleBuilder::module_graph_;
+std::map<std::string, Node> ModuleBuilder::task_graph_;
+
+int ModuleBuilder::AddEdge(const std::string &from, const std::string &to) {
+  auto it = module_graph_.find(from);
+  if (it == module_graph_.end()) {
+    return -EINVAL;
+  }
+  it->second.children().insert(to);
+  UpdateTCGraph();
+  return 0;
+}
+
+int ModuleBuilder::UpdateTCGraph() {
+  int ret = 0;
+  for (auto const &tc : task_graph_) {
+    std::unordered_set<std::string> visited;
+    ret = FindNextTC(tc.first, tc.first, visited);
```

Comment:
Perhaps rewrite this and the next few lines as:

`
if (!FindNextTC(...)) {
  return false;
}
`